drivers/sensor/sht4x/sht4x.c

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright (c) 2021 Leonard Pollak
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT sensirion_sht4x
+
+#include <device.h>
+#include <drivers/i2c.h>
+#include <kernel.h>
+#include <drivers/sensor.h>
+#include <sys/__assert.h>
+#include <logging/log.h>
+#include <sys/byteorder.h>
+#include <sys/crc.h>
+
+#include <drivers/sensor/sht4x.h>
+#include "sht4x.h"
+
+LOG_MODULE_REGISTER(SHT4X, CONFIG_SENSOR_LOG_LEVEL);
+
+static uint8_t sht4x_compute_crc(uint16_t value)
+{
+	uint8_t buf[2];
+
+	sys_put_be16(value, buf);
+
+	return crc8(buf, 2, SHT4X_CRC_POLY, SHT4X_CRC_INIT, false);
+}
+
+static int sht4x_write_command(const struct device *dev, uint8_t cmd)
+{
+	const struct sht4x_config *cfg = dev->config;
+	uint8_t tx_buf[1] = { cmd };
+
+	return i2c_write(cfg->bus, tx_buf, sizeof(tx_buf), cfg->i2c_addr);
+}
+
+static int sht4x_read_sample(const struct device *dev,
+		uint16_t *t_sample,
+		uint16_t *rh_sample)
+{
+	const struct sht4x_config *cfg = dev->config;
+	uint8_t rx_buf[6];
+	int rc;
+
+	rc = i2c_read(cfg->bus, rx_buf, sizeof(rx_buf), cfg->i2c_addr);
+	if (rc) {
+		LOG_ERR("Failed to read data from device.");
+		return rc;
+	}
+
+	*t_sample = sys_get_be16(rx_buf);
+	if (sht4x_compute_crc(*t_sample) != rx_buf[2]) {
+		LOG_ERR("Invalid CRC for T.");
+		return -EIO;
+	}
+
+	*rh_sample = sys_get_be16(&rx_buf[3]);
+	if (sht4x_compute_crc(*rh_sample) != rx_buf[5]) {
+		LOG_ERR("Invalid CRC for RH.");
+		return -EIO;
+	}
+
+	return 0;
+}
+
+/* public API for handling the heater */
+int sht4x_fetch_with_heater(const struct device *dev)
+{
+	struct sht4x_data *data = dev->data;
+	int rc;
+
+	rc = sht4x_write_command(dev,
+			heater_cmd[data->heater_power][data->heater_duration]);
+	if (rc) {
```

Comment:
nit: in order to avoid int boolean checks, better use `if (rc < 0)` everywhere (ref. MISRA C:2012 Rule 14.4)

Reply:
i've updated this as well as the sgp40.c
sed ftw